test/model/message_list_test.dart

```diff
@@ -424,6 +425,355 @@ void main() {
     });
   });
 
+  group('messagesMoved', () {
+    final stream = eg.stream(streamId: 1, name: 'test stream');
+    final otherStream = eg.stream(streamId: 2, name: 'other stream');
+
+    void checkHasMessages(Iterable<Message> messages) {
+      check(model.messages.map((e) => e.id)).deepEquals(messages.map((e) => e.id));
+    }
+
+    Future<void> prepareNarrow(Narrow narrow, List<Message> messages) async {
+      await prepare(narrow: narrow);
+      for (final streamToAdd in [stream, otherStream]) {
+        final subscription = eg.subscription(streamToAdd);
+        await store.addStream(streamToAdd);
+        await store.addSubscription(subscription);
+      }
+      if (messages.isNotEmpty) {
+        await prepareMessages(foundOldest: false, messages: messages);
+      }
+      checkHasMessages(messages);
+    }
+
+    group('in topic narrow', () {
+      final narrow = TopicNarrow(stream.streamId, 'topic');
+      final initialMessages = List.generate(5, (i) => eg.streamMessage(stream: stream, topic: 'topic'));
+      final movedMessages = List.generate(5, (i) => eg.streamMessage(stream: stream, topic: 'topic'));
+
+      test('internal move', () async {
+        await prepareNarrow(narrow, initialMessages + movedMessages);
+
+        await store.handleEvent(eg.updateMessageMoveEvent(movedMessages,
+          newTopic: movedMessages[0].topic,
+          origTopic: movedMessages[0].topic
```

Comment:
I wonder how this is possibly a move. Does internal move within a topic narrow actually happen?

Reply:
Hmm. Yeah, this case seems like it should be a no-op and shouldn't happen. So maybe it should just look like this (in the implementation):
```dart
          case (true,  true ): return; // TODO(log) no-op move
```